### DISCUSSION TITLE
GitHub readme stats (Top languages) not working properly

### DISCUSSION THREAD

**Discussion by Riyakumari57** [anuraghazra/github-readme-stats#3083] [Q&A] [answered]
Why do these two differ
[https://github-readme-stats.vercel.app/api/top-langs/?username=Riyakumari57&langs_count=10&theme=radical](https://github.com/anuraghazra/github-readme-stats/issues/url)
[http://ionicabizau.github.io/github-profile-languages/?user=Riyakumari57](https://github.com/anuraghazra/github-readme-stats/issues/url)

the first link which i am using for my gitHub profile shows 42% c++ which is on the top but in the second link javascript section area covers the major part (44.44%) and c++ is 11.11%

**Comment by qwerty541** [comment, marked as answer]
Hey, @Riyakumari57!

http://ionicabizau.github.io/github-profile-languages/ Uses under the hood [gh-polyglot](https://www.npmjs.com/package/gh-polyglot) package. Languages usage calculation inside this package based on how many user repositories contains some specific programming language as main. You can check it on the following lines inside their sources:

https://github.com/IonicaBizau/node-gh-polyglot/blob/0849accbadf769a5cd59b42a01c47ba5e6c638ec/lib/index.js#L103-L106

https://github-readme-stats.vercel.app/ Works directly with GraphQL GitHub API under the hood. By default languages usage calculation based on bytes count. You can read more about our languages percentage calculation algorithm in our documentation: https://github.com/anuraghazra/github-readme-stats#language-stats-algorithm

![image](https://github.com/anuraghazra/github-readme-stats/assets/53787217/a4294666-afe8-4949-b21d-e71d2e109c64)
